reana_cluster/cli/cluster.py

```diff
@@ -0,0 +1,211 @@
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA cluster administration (init, restart, etc.) commands."""
+
+
+import logging
+
+import click
+import os
+
+import errno
+import yaml
+
+from ..config import generated_cluster_conf_default_path
+
+from reana_cluster import KubernetesBackend
+
+from ..utils import load_spec_file
+
+
+@click.command()
+@click.option('--remove-persistent-storage', is_flag=True,
+              help='If set, also persistent storage '
+                   'inside the cluster is deleted.')
+@click.pass_context
+def down(ctx, remove_persistent_storage):
+    """Bring REANA cluster down, i.e. shutdown all components."""
+
+    try:
+        cluster_spec = load_spec_file(click.format_filename(
+            ctx.parent.params['file']),
+            ctx.parent.params['skip_validation'])
+
+        backend = KubernetesBackend(cluster_spec)
+
+        backend.down()
+
+    except Exception as e:
+        logging.debug(str(e))
+
+
+@click.command()
+# @click.argument('component', help='Name of the component which URL should '
+#                                   'be resolved.')
+@click.argument('component')
+# @click.option('--namespace', help='Namespace of the component which URL '
+#                                   'should be resolved.')
+@click.option('--namespace')
+@click.pass_context
+def get(ctx, component, namespace):
+    """Fetch URL for a REANA component running in REANA cluster."""
+
+    try:
+        cluster_spec = load_spec_file(click.format_filename(
+            ctx.parent.params['file']),
+            ctx.parent.params['skip_validation'])
+
+        backend = KubernetesBackend(cluster_spec)
+
+        click.echo(backend.get_component(component))
+
+    except Exception as e:
+        logging.debug(str(e))
+
+
+@click.command()
+@click.option('--remove-persistent-storage', is_flag=True,
+              help='If set, also persistent storage '
+                   'inside the cluster is deleted.')
+@click.pass_context
+def restart(ctx, remove_persistent_storage):
+    """Restart all components running in REANA cluster."""
+    try:
+        cluster_spec = load_spec_file(click.format_filename(
+            ctx.parent.params['file']),
+            ctx.parent.params['skip_validation'])
+
+        backend = KubernetesBackend(cluster_spec)
+
+    except Exception as e:
+        logging.debug(str(e))
+
+
+@click.command()
+@click.option('--skip-initialization', is_flag=True,
```

Comment:
is this better named as `skip-initialization` or as `dry-run`? what do you think?

Reply:
Not sure.